### ISSUE TITLE
Trying to get content fail 

### ISSUE THREAD

**Issue by mathieukempe** [octokit/octokit.net#1111]
Using 
https://developer.github.com/v3/repos/contents/

This used to work with 0.10.0

<pre>
var api = new ApiConnection(new Connection(new ProductHeaderValue("something"), new InMemoryCredentialStore(new Credentials(_token))));

 string url = string.Format("https://api.github.com/repos/{0}/{1}/contents/{2}?ref={3}", owner, repo, path, branch);
            
 var blob = api.Get<Octokit.Blob>(new Uri(url)).Result;
</pre>

Using the same call with curl I can retrieve the content of file but using octokit 0.18.0 api.Get<Octokit.Blob>(new Uri(url)).Result just hang forever.

**Comment by shiftkey** [comment]
@mathieukempe not sure whether this is an outstanding issue, but here's how I'd do this without using `ApiConnection` directly (this is using 0.18.0):

```csharp
var client = new GitHubClient("something");
client.Credentials = new Credentials(_token);
var contents = await client.Repository.Content.GetAllContents(owner, repo, path, branch)
var file = contents[0].Content;
```

**Comment by mathieukempe** [comment]
@shiftkey thanks for the example of code
The issue seems to be when I call the code in the controller asp.net mvc 5

```c#
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            var client = new GitHubClient(new ProductHeaderValue("something"), new InMemoryCredentialStore(new Credentials("my api key")));

            var content = client.Repository.Content.GetAllContents("Selz", "drachma",
                "src/Drachma.Web/Resources/Buttons.resx").Result;



            return View();
        }

```
The call to GetAllContents hang
The same code in a console app work.

**Comment by shiftkey** [comment]
@mathieukempe to be honest, I'm not sure how this might be deadlocking.

We have the `ConfigureAwait` setup [inside GetAll](https://github.com/octokit/octokit.net/blob/master/Octokit/Http/ApiConnection.cs#L147-L148) so perhaps making this method `async/await` might change the behaviour (rather than blocking using `.Result`).

**Comment by zzzprojects** [comment]
There is a lot of ConfigureAwait(false) missing in all the projects.

By example, the method "client.Repository.Content.GetAllContents(...)" didn't work in my WinForm application because of UI Blocking.

I just fixed RedirectHandler.SendAsync to add ConfigureAwait to all methods and it worked.

**Comment by shiftkey** [comment]
@zzzprojects thanks for the extra info - I've opened #1248 which cleans this up

**Comment by ryangribble** [comment]
This was fixed by #1248 (apart from a few additional ones which were possibly introduced after #1248 anyway... #1483 is tackling those)
